packages/cdk/lib/roles/cromwell-adapter-role.ts

```diff
@@ -0,0 +1,21 @@
+import * as cdk from "monocdk";
+import * as iam from "monocdk/aws-iam";
+import { NextflowDescribeJobsBatchPolicy } from "./policies/nextflow-describe-jobs-batch-policy";
+import { NextflowSubmitJobBatchPolicy } from "./policies/nextflow-submit-job-batch-policy";
+import { BucketOperations } from "../../common/BucketOperations";
+
+export interface CromwellAdapterRoleProps {
+  readOnlyBucketArns: string[];
+  readWriteBucketArns: string[];
+}
+
+export class CromwellAdapterRole extends iam.Role {
+  constructor(scope: cdk.Construct, id: string, props: CromwellAdapterRoleProps) {
+    super(scope, id, {
+      assumedBy: new iam.ServicePrincipal("ecs-tasks.amazonaws.com"),
+    });
+
+    BucketOperations.grantBucketAccess(this, this, props.readOnlyBucketArns, true);
```

Comment:
Why adapter needs access to S3 buckets? Is it not just pass though requests from a client to an engine?

Reply:
The Cromwell adapter reads workflow bundles from S3. At minimum, it will need access to a context's S3 location.

Reply:
Does it need access to s3 locations defined in data section of the project?

Reply:
I tested this by running bam-unmapped-to-mapped with the permissions above and it passed successfully, which uses inputs from other S3 buckets. I am not sure about how the WES Adapter works, but in my experimentation, both nextflow and cromwell adapters need access to the context's s3 location or workflow execution fails.